File: Section 2/2.5 False Position/false_position.c
#include <stdio.h>
#include <math.h>

double f(double x){
	/* === DEFINE THE FUCTION TO USE ===*/
	return 0;
}

double FalsePosition(double p_0, double p_1, int n, double tol){
	double p, q;
	double q_0 = f(p_0);
	double q_1 = f(p_1);

	int i = 2;
	while (i < n){
		p = p_1 - (q_1 * (p_1 - p_0) / (q_1 - q_0));

		if (fabs(p - p_1) < tol)
			return p;

		i++;
		q = f(p);

		if ((q * q_1) < 0){
			p_0 = p_1;
			q_0 = q_1;
		}

		p_1 = p;
		q_1 = q;
	}
	printf("The method failed after %d iterations.\n", n);
}

int main(){
	/* === INSERT DATA HERE === */
	double n;
	double p_0;
	double p_1;
	double tol;

	/* === CALCULATE === */
	double r = FalsePosition(p_0, p_1, n, tol);
	printf("%lf\n", r);
	return 0;
}

File: 1. SOLUCION DE ECUACIONES/5. Metodo de la falsa posicion/false_position.m
function r = falseposition(f, p_0, p_1, n, tol)
	q_0 = f(p_0);
	q_1 = f(p_1);

	i = 2;
	while i < n
		p = p_1 - (q_1 * (p_1 - p_0) / (q_1 - q_0));

		if (abs(p - p_1) < tol)
			r = p;
			return;
		endif

		i++;
		q = f(p);

		if (q*q_1) < 0
			p_0 = p_1;
			q_0 = q_1;
		endif

		p_1 = p;
		q_1 = q;
	endwhile
	printf("The method failed after %d iterations.\n", n);
endfunction
